custom_components/solplanet/services.py

```diff
@@ -0,0 +1,162 @@
+
+        isns = await get_isn_from_target(hass, target)
+        if not isns:
+            raise vol.Invalid("No valid entities or devices found")
+            
+        processed = False
+        for isn in isns:
+            for entry_id, data in hass.data[DOMAIN].items():
+                coordinator = data["coordinator"]
+                if isn in coordinator.data[BATTERY_IDENTIFIER]:
+                    try:
+                        # Get current schedule
+                        current_schedule = coordinator.data[BATTERY_IDENTIFIER][isn]["schedule"]["slots"]
+
+                        # Check if we already have 6 slots for this day
+                        if call.data["day"] in current_schedule and len(current_schedule[call.data["day"]]) >= 6:
+                            raise vol.Invalid("Cannot add more than 6 slots per day")
+                        
+                        # Create new slot
+                        slot = ScheduleSlot.from_time(
+                            start=f"{call.data['start_hour']:02d}:{call.data['start_minute']:02d}",
+                            duration=call.data["duration"],
+                            mode=call.data["mode"]
+                        )
+                        
+                        # Get existing slots for the day
+                        new_slots = dict(current_schedule)
+                        day_slots = new_slots.get(call.data["day"], [])
+                        day_slots.append(slot)
+                        
+                        try:
+                            # This will validate max slots, overlaps, and midnight crossing
+                            ScheduleSlot.validate_slots(day_slots)
+                        except ValueError as err:
+                            raise vol.Invalid(str(err)) from err
+                            
+                        new_slots[call.data["day"]] = day_slots
+                        
+                        await coordinator.set_battery_schedule_slots(isn, new_slots)
+                        processed = True
+                        break
+                    except Exception as err:
+                        raise
+                    
+        if not processed:
+            raise vol.Invalid(f"No valid battery coordinator found for ISNs {isns}")
+
+    async def clear_schedule(call: ServiceCall) -> None:
+        """Handle clear_schedule service."""
+        target = call.target if hasattr(call, 'target') else {}
+        if 'entity_id' in call.data:
+            target['entity_id'] = call.data['entity_id']
+        if 'device_id' in call.data:
+            target['device_id'] = call.data['device_id']
+
+        isns = await get_isn_from_target(hass, target)
+        if not isns:
+            raise vol.Invalid("No valid entities or devices found")
+            
+        processed = False
+        for isn in isns:
+            for entry_id, data in hass.data[DOMAIN].items():
```

Comment:
`entry_id` is not used

```suggestion
            for data in hass.data[DOMAIN].values():
```